### PR TITLE
Added PySpark program plugin

### PR DESCRIPTION
**Pull request by chtyim**
Also removed unnecessary `SparkDataFrames` class, which is provided by cdap-api since 4.3.0

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -261,82 +268,20 @@
...
       <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-antrun-plugin</artifactId>
-        <version>1.7</version>
+        <groupId>co.cask</groupId>
+        <artifactId>cdap-maven-plugin</artifactId>
+        <version>1.0.1-SNAPSHOT</version>
+        <configuration>
+          <cdapArtifacts>
+            <parent>system:cdap-data-pipeline[4.3.0-SNAPSHOT,9.0.0-SNAPSHOT)</parent>
```
Comment (albertshau) [inline pom.xml]: we probably shouldn't make this go up to 9.0.0. Maybe have 5.0.0-SNAPSHOT as the upper bound so it doesn't cross a major version.

File: src/main/java/co/cask/hydrator/plugin/spark/dynamic/PySparkProgram.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  /**
+   * Plugin configuration class.
+   */
+  public static final class Config extends PluginConfig {
+
+    @Description(
+      "The source code of the PySpark program written in Python. " +
+        "The content must be a valid Python code.")
```
Comment (albertshau) [inline src/main/java/co/cask/hydrator/plugin/spark/dynamic/PySparkProgram.java]: 'must be a' -> 'must be'

File: widgets/PySparkProgram-sparkprogram.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "metadata": {
+    "spec-version": "1.0"
```
Comment (albertshau) [inline widgets/PySparkProgram-sparkprogram.json]: i think this is supposed to be some higher version (1.4? don't remember). Unless the UI supports new features (icon) even with an older spec version?

File: widgets/ScalaSparkCompute-sparkcompute.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "metadata": {
-    "spec-version": "1.5"
+    "spec-version": "1.4"
```
Comment (bdmogal) [inline widgets/ScalaSparkCompute-sparkcompute.json:3]: this should be 1.5 I think. ``display-name`` and ``icon`` are supported in 1.5.

File: pom.xml [hunk 5]
```diff
@@ -273,7 +273,7 @@
         <version>1.0.1-SNAPSHOT</version>
         <configuration>
           <cdapArtifacts>
-            <parent>system:cdap-data-pipeline[4.3.0-SNAPSHOT,9.0.0-SNAPSHOT)</parent>
+            <parent>${app.parents}</parent>
```
Comment (albertshau) [inline pom.xml]: does this produce a config with two element in the parents json array? I thought multiple parents are supposed to be specified as multiple 'parent' elements in this section.

Comment (chtyim) [reply pom.xml]: oh yeah.. so need to have two `<parent>`?

Comment (chtyim) [reply pom.xml]: hum.. meaning I can't easily define it in the property.

**Review by albertshau — APPROVED** [review]
lgtm

File: widgets/ScalaSparkCompute-sparkcompute.json (hunk 4 above)
Comment (albertshau) [reply widgets/ScalaSparkCompute-sparkcompute.json:3]: ajai mentioned it is 1.4. But it seems like these versions really haven't needed to be incremented at all. There are no backwards incompatibilities and new features not present in older specs work anyway.

**Comment by chtyim** [comment]
Squashed and merging.

**Pull request merged**